src/org/thoughtcrime/securesms/map/EmojiBitmapGenerationTask.java

```diff
@@ -0,0 +1,56 @@
+package org.thoughtcrime.securesms.map;
+
+import android.graphics.Bitmap;
+import android.os.AsyncTask;
+
+import org.thoughtcrime.securesms.components.emoji.EmojiProvider;
+import org.thoughtcrime.securesms.util.Pair;
+
+import java.lang.ref.WeakReference;
+
+public class EmojiBitmapGenerationTask extends AsyncTask<String, Pair<String, Bitmap>, Void> {
```

Comment:
https://developer.android.com/reference/android/os/AsyncTask says AsyncTask is deprecated.

Given that progress update and other features are not used, maybe use a simple thread like suggested in https://stackoverflow.com/a/63500657 ?

Reply:
Dang! Thanks for the hint. It makes to not implement new stuff using the pattern.

Reply:
It's used, but only to return the result once.

Reply:
no, look at line 32, it's in a loop and it's the method to invoke onProgressUpdate()

Reply:
Anyways the whole class vanished in the meanwhile